src/main/java/seedu/clinic/commons/util/StringUtil.java

```diff
@@ -60,7 +60,25 @@ public static boolean isNonZeroUnsignedInteger(String s) {
 
         try {
             int value = Integer.parseInt(s);
-            return value > 0 && !s.startsWith("+"); // "+1" is successfully parsed by Integer#parseInt(String)
+            return value != 0 && isUnsignedInteger(s);
+        } catch (NumberFormatException nfe) {
+            return false;
+        }
+    }
+
+    /**
+     * Returns true if {@code s} represents a non-zero unsigned integer
+     * e.g. 1, 2, 3, ..., {@code Integer.MAX_VALUE} <br>
+     * Will return false for any other non-null string input
+     * e.g. empty string, "-1", "0", "+1", and " 2 " (untrimmed), "3 0" (contains whitespace), "1 a" (contains letters)
+     * @throws NullPointerException if {@code s} is null.
+     */
+    public static boolean isUnsignedInteger(String s) {
+        requireNonNull(s);
+
+        try {
+            int value = Integer.parseInt(s);
+            return value >= 0 && !s.startsWith("+"); // "+1" is successfully parsed by Integer#parseInt(String)
```

Comment:
Would this mean that 0 is acceptable? Or is `isNonZeroUnsignedInteger` meant to be a stricter check?

Reply:
Oh I see that it is being used for index and quantity separately in another file.

Reply:
Yes, It is the case. the latter is the stricter check.

Reply:
Alright got it!